android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -359,9 +360,19 @@ constructor(
       list.filter { getValue(it, attribute) == attributeValue }
 
     private fun getValue(resource: Resource, attribute: String): Any? {
-      val property = (resource.javaClass).getDeclaredField(attribute)
-      property.isAccessible = true
-      return (property.get(resource) as? PrimitiveType<*>)?.value
+      val property = getDeclaredField(attribute, resource.javaClass)
+      property?.isAccessible = true
+      return (property?.get(resource) as? PrimitiveType<*>)?.value
+    }
+
+    private fun getDeclaredField(attribute: String, type: Class<*>): Field? {
+      return try {
+        type.getDeclaredField(attribute)
+      } catch (ex: Exception) {
+        if (type.superclass != null) {
+          getDeclaredField(attribute, type.superclass)
+        } else null
```

Comment:
do we want to raise instead or returning null here?

Reply:
returning `null` because might be a chance the field that we want is not available in the entire class hierarchy and `null` would be a case of empty value in the application.

Reply:
cool, great explanation, I moved that into a comment

Reply:
@owais-vd would we want to check `getDeclaredField(attribute, type.superclass)` first and then if that throws the exception we catch it and then try the `type.getDeclaredField(attribute)`?

Reply:
sorry, I'm not getting your point, why do we go superclass first? could you add more clarification? Thanks

Reply:
oh I see, I thought the second parameter for getDeclaredField was to strongly type the value returned, such that if the field is a resource like patient and the type is Patient then we want to evaluate that first. If it fails then we fall back to the one without type.